### PR TITLE
Use CDS components on item report page

### PR DESCRIPTION
**Pull request by brianfleming**
## Description

- Improve the styles on the item report page per CDS
- remove the `LimitedTextFieldWithCounter` component now that its redundant with another CDS component
- remove `ReportDesignerFormSection.js` since it was only used in one place
- Change out the relay container loader for the meedan animated loader
- Add a page level banner to explain how to enable editing the report when its paused

## Type of change

- [x] Performance improvement and/or refactoring (non-breaking change that keeps existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Security mitigation or enhancement
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Automated test (add or update automated tests)

### AFTER Text
Screenshot 2023-12-11 at 11 03 08 AM

### AFTER Visual
Screenshot 2023-12-11 at 11 03 21 AM

### AFTER No Intro
Screenshot 2023-12-11 at 11 05 16 AM

### AFTER disabled
Screenshot 2023-12-11 at 1 03 22 PM

### AFTER pause banner
Screenshot 2023-12-11 at 1 45 04 PM

## Checklist
- [x] I have performed a self-review of my own code
- [x] I've made sure my branch is runnable and given good testing steps in the PR description
- [x] I considered secure coding practices when writing this code. Any security concerns are noted above.
- [x] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I implemented any new components, they are self-contained, their `propTypes` are declared and they use React Hooks and, if data-fetching is required, they use Relay Modern with fragment containers
- [ ] If my components involve user interaction - specifically button, text fields, or other inputs - I have added a [BEM-like class name](https://meedan.atlassian.net/wiki/spaces/ENG/pages/1327628289/Naming+conventions+for+interactive+elements) to the element that is interacted with
- [x] To the best of my knowledge, any new styles are applied according to the design system
- [ ] If I added a new external dependency, I included a rationale for doing so and an estimate of the change in bundle size (e.g., checked in https://bundlephobia.com/)

### CODE REVIEW

**Comment by brianfleming** [comment]
Integration test failure looks like the "typical" 4 that are unrelated to this PR

**Comment by brianfleming** [comment]
just a friendly review reminder please

**Comment by amoedoamorim** [comment]
Reviewing this

File: src/app/components/media/ReportDesigner/ReportDesignerForm.js [hunk 1]
```diff
@@ -90,292 +55,490 @@ const ReportDesignerForm = (props) => {
...
+                  key="false"
+                  className="int-report__button--report-type"
+                >
+                  <FormattedMessage id="reportDesigner.lightTheme" defaultMessage="Light" description="Label used for button toggling on the light themed report" />
+                </ToggleButton>
+                <ToggleButton
+                  selected={Boolean(data.dark_overlay)}
+                  value={Boolean(true)}
+                  key="true"
+                  className="int-report__button--report-type"
```
Comment (amoedoamorim) [inline src/app/components/media/ReportDesigner/ReportDesignerForm.js:327]: It looks like these classNames are repeated. If we're using them for automated tests, how about changing them to something like `int-report__button--theme-dark` and `int-report__button--theme-light` ?

File: src/app/components/media/ReportDesigner/ReportDesignerForm.js [hunk 2]
```diff
@@ -90,292 +55,490 @@ const ReportDesignerForm = (props) => {
...
+                    onChange={handleChangeColor}
+                  />
+                  <FormattedMessage
+                    id="reportDesigner.textStatusPlaceholder"
+                    defaultMessage="Add a status to this report"
+                    description="Placeholder for report status field"
+                  >
+                    { placeholder => (
+                      <LimitedTextArea
+                        required={Boolean(false)}
```
Comment (amoedoamorim) [inline src/app/components/media/ReportDesigner/ReportDesignerForm.js:345]: Not really the scope of this PR, but I just wanted to vent that required as default is annoying :smiling_face_with_tear:

**Review by amoedoamorim — APPROVED** [review]
PR looks good. Please see my suggestion about diff classNames for theme toggle buttons

File: src/app/components/media/ReportDesigner/ReportDesignerForm.js (hunk 2 above)
Comment (brianfleming) [reply src/app/components/media/ReportDesigner/ReportDesignerForm.js:345]: yeah...when I ran into this I thought the same thing. maybe next time i do a pure component improvement pass to match the design system I will tick this off

**Pull request merged**